index/src/executors/bridge_executors/mod.rs

```diff
@@ -20,9 +20,9 @@ use scale::Encode;
 use subrpc::{create_transaction, send_transaction, ExtraParam};
 use xcm::v1::{prelude::*, AssetId, Fungibility, Junction, Junctions, MultiAsset, MultiLocation};
 
-mod moonbeam_to_acala;
-mod moonbeam_to_phala;
-mod phala_to_acala;
+pub mod moonbeam_to_acala;
+pub mod moonbeam_to_phala;
+pub mod phala_to_acala;
 
 #[derive(Clone)]
 pub struct ChainBridgeEvm2Phala {
```

Comment:
Is the `new()` function called for every query (when you init all the executors)? Parsing ABI will be very expensive. So I suggest to implement it as a lazy operation instead. We should only load the ABI when the contract will be called.

Reply:
Agree, I will have a try

Reply:
Now only be called once per query operatioin